src/shell/doom.c

```diff
@@ -96,6 +97,17 @@ static const char *doom_exit_host (struct shell_doom *doom)
     return hostlist_nth (doom->hl, nth);
 }
 
+static char *get_jobspec_command_arg0 (struct shell_doom *doom)
+{
+    static char buf[PATH_MAX+1] = {0};
+    json_t *s = json_array_get (doom->shell->info->jobspec->command, 0);
+    const char *path = json_string_value (s);
+    char *rv;
+    snprintf (buf, PATH_MAX, "%s", path);
+    rv = basename (buf);
```

Comment:
Would it be simpler to just do something like
```c
const char *p = strrchr (s, "/");
const char *base = p ? p + 1 : s;
```

And avoid the copy? Maybe we should have our own implementation of `basename()` that doesn't modify its argument (I think some places in the codebase may already assume GNU `basename(3)`, which doesn't modify its arg, but since we support alpine and someday BSD, that's probably not safe).

Reply:
yeah, that would be better.  When I read the manpage for basename(3) I was like "huh, it needs to modify the buffer?"  Lets go with your simpler approach.